dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java

```diff
@@ -364,63 +369,147 @@ protected ModifyJobPriorityModel call() throws Exception {
                 }
-
-                final List<FileTreeModel> selectedItemsAtDestinationList = selectedItemsAtDestination.stream().map(TreeItem::getValue).collect(Collectors.toList());
+                final List<FileTreeModel> selectedItemsAtDestinationList = selectedItemsAtDestination.stream()
+                        .map(TreeItem::getValue).collect(Collectors.toList());
                 @SuppressWarnings("unchecked")
                 final TreeTableView<Ds3TreeTableValue> ds3TreeTableView = getTreeTableView();
-                final ImmutableList<TreeItem<Ds3TreeTableValue>> selectedItemsAtSourceLocation = ds3TreeTableView.getSelectionModel().getSelectedItems()
+                ImmutableList<TreeItem<Ds3TreeTableValue>> selectedItemsAtSourceLocation = ds3TreeTableView.getSelectionModel().getSelectedItems()
                         .stream().collect(GuavaCollectors.immutableList());
-
-                if (selectedItemsAtSourceLocation.isEmpty()) {
-                    LOG.error("Files not selected");
-                    ALERT.setContentText("Please select files to transfer");
+                final TreeItem<Ds3TreeTableValue> root = ds3TreeTableView.getRoot();
+                if (selectedItemsAtSourceLocation.isEmpty() && root == null) {
+                    LOG.info("Files not selected");
+                    ALERT.setContentText(resourceBundle.getString("fileSelectError"));
                     ALERT.showAndWait();
                     return;
+                } else if (selectedItemsAtSourceLocation.isEmpty()) {
+                    final ImmutableList.Builder<TreeItem<Ds3TreeTableValue>> builder = ImmutableList.builder();
+                    selectedItemsAtSourceLocation = builder.add(root).build().asList();
                 }
-
-                final List<Ds3TreeTableValue> selectedItemsAtSourceLocationList = selectedItemsAtSourceLocation.stream().map(TreeItem::getValue).collect(Collectors.toList());
-                final List<Ds3TreeTableValueCustom> selectedItemsAtSourceLocationListCustom = selectedItemsAtSourceLocationList.stream().map(v -> new Ds3TreeTableValueCustom(v.getBucketName(), v.getFullName(), v.getType(), v.getSize(), v.getLastModified(), v.getOwner(), v.isSearchOn())).collect(Collectors.toList());
-
+                //Getting selected item at source location
+                final List<Ds3TreeTableValue> selectedItemsAtSourceLocationList = selectedItemsAtSourceLocation.stream()
+                        .map(TreeItem::getValue).collect(Collectors.toList());
+                final List<Ds3TreeTableValueCustom> selectedItemsAtSourceLocationListCustom =
+                        selectedItemsAtSourceLocationList.stream()
+                                .map(v -> new Ds3TreeTableValueCustom(v.getBucketName(),
+                                        v.getFullName(), v.getType(), v.getSize(), v.getLastModified(),
+                                        v.getOwner(), v.isSearchOn())).collect(Collectors.toList());
```

Comment:
Unless the List that is returned by the map statement needs to be modified, this should use the Guava ImmutableList collector